airflow/providers/google/cloud/hooks/dataprep.py

```diff
@@ -50,26 +54,60 @@ def _headers(self) -> Dict[str, str]:
         }
         return headers
 
-    @property
-    def _token(self) -> str:
-        conn = self.get_connection(self.dataprep_conn_id)
-        token = conn.extra_dejson.get("token")
-        if token is None:
-            raise AirflowException(
-                "Dataprep token is missing or has invalid format. "
-                "Please make sure that Dataprep token is added to the Airflow Connections."
-            )
-        return token
-
     @retry(stop=stop_after_attempt(5), wait=wait_exponential(multiplier=1, max=10))
     def get_jobs_for_job_group(self, job_id: int) -> Dict[str, Any]:
         """
         Get information about the batch jobs within a Cloud Dataprep job.
 
-        :param job_id The ID of the job that will be fetched.
+        :param job_id The ID of the job that will be fetched
         :type job_id: int
         """
-        url: str = f"{self._url}/{job_id}/jobs"
+
+        endpoint_path = f"/v4/jobGroups/{job_id}/jobs"
+        url: str = os.path.join(self._base_url, endpoint_path)
         response = requests.get(url, headers=self._headers)
         response.raise_for_status()
```

Comment:
I remember there was a problem with error message propagation - only code was returned instead of error message. Is this still an issue?

Reply:
It would be nice to fix it because running the system test I get:
```
Traceback (most recent call last):
  File "/usr/local/lib/python3.7/site-packages/tenacity/__init__.py", line 362, in call
    result = fn(*args, **kwargs)
  File "/opt/airflow/airflow/providers/google/cloud/hooks/dataprep.py", line 112, in run_job_group
    response.raise_for_status()
  File "/usr/local/lib/python3.7/site-packages/requests/models.py", line 941, in raise_for_status
    raise HTTPError(http_error_msg, response=self)
requests.exceptions.HTTPError: 404 Client Error: Not Found for url: https://api.clouddataprep.com/v4/jobGroups
```
and I'm quite sure that during debugging we have seen a nice error message

Reply:
```python
    def _raise_for_status(self, response):
        try:
            response.raise_for_status()
        except HTTPError:
            self.log.error(response.json().get('exception'))
            raise
```
this may help

Reply:
That's great.